### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by dicodiaz**
- Implement thunks to consume API.
- Add BookstoreAPI class to manage the API consumption.
- Refactor Components.

I take into account the following requirements:
- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- Refactor your *add book* and *remove book* features to persist your changes in the server.
  - You should refactor the redux code of your application using *middleware* to make async requests to the API.
  - Don't do the changes using only React state.
- *Add book* and *remove book* should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @dicodiaz 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "react-scripts": "4.0.3",
     "redux": "^4.1.2",
     "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.4.0",
```
Comment (mikenath223) [inline package.json:21]: - [x] Kindly deploy the project online and include the live link in the readme.

**Comment by mikenath223** [comment]
- [ ] In eslintrc.json you seem to have added more rules than stated [here](https://github.com/microverseinc/linters-config/tree/master/react-redux), please remove all extra rules.

File: package.json (hunk 1 above)
Comment (dicodiaz) [reply package.json:21]: Hello @mikenath223, I will do it, but can you kindly explain why this is a requirement?

**Comment by dicodiaz** [comment]
> * [x]  In eslintrc.json you seem to have added more rules than stated [here](https://github.com/microverseinc/linters-config/tree/master/react-redux), please remove all extra rules.

Hello @mikenath223. You're right, I did add some extra rules, following the official React [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html) documentation. This was referenced in a Microverse lesson. Also, this means that my project follows these new rules **on top of** Microverse suggested linters, so I'm not by any means ignoring Microverse linters. Can you kindly explain why I should remove them?

File: package.json (hunk 1 above)
Comment (dicodiaz) [reply package.json:21]: Please have in mind that I deployed to GitHub Pages, as referenced in the README.

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @dicodiaz, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**